src/components/dashboard/main-content/course-enrollments/course-cards/BaseCourseCard.jsx

```diff
@@ -158,6 +181,65 @@ class BaseCourseCard extends Component {
 
   resetModals = () => {
     this.setModalState({ key: 'emailSettings' });
+    this.setModalState({ key: 'unenroll' });
+  };
+
+  handleUnenrollButtonClick = () => {
+    const { courseRunId } = this.props;
+    const { enterpriseConfig } = this.context;
+    this.setModalState({
+      key: 'unenroll',
+      open: true,
+    });
+    sendEnterpriseTrackEvent(
+      enterpriseConfig.uuid,
+      'edx.ui.enterprise.learner_portal.dashboard.enrollments.course.unenroll_modal.opened',
+      { course_run_id: courseRunId },
+    );
+  };
+
+  renderUnenrollModal = () => {
+    const {
+      canUnenroll, courseRunId, title, type,
+    } = this.props;
+    const { modals } = this.state;
+
+    if (!canUnenroll) {
+      return null;
+    }
+
+    return (
+      <UnenrollModal
+        courseRunId={courseRunId}
+        courseRunTitle={title}
+        onClose={this.handleUnenrollModalOnClose}
+        onSuccess={this.handleUnenrollModalSuccess}
+        isOpen={modals.unenroll.open}
+        enrollmentType={camelCase(type)}
+      />
+    );
+  };
+
+  handleUnenrollModalOnClose = () => {
+    this.resetModals();
+    const { courseRunId } = this.props;
+    const { enterpriseConfig } = this.context;
+    sendEnterpriseTrackEvent(
+      enterpriseConfig.uuid,
+      'edx.ui.enterprise.learner_portal.dashboard.enrollments.course.unenroll_modal.closed',
+      { course_run_id: courseRunId },
+    );
+  };
+
+  handleUnenrollModalSuccess = () => {
```

Comment:
galactic nit for consistency, `handleUnenrollModalSuccess` -> `handleUnenrollModalOnSuccess` or remove On from `handleUnenrollModalOnClose`